pantheon-bundle/src/main/java/com/redhat/pantheon/asciidoctor/AsciidoctorService.java

```diff
@@ -153,30 +151,38 @@ private String buildModule(Module base, ModuleVersion moduleVersion, Map<String,
 
             // process product and version.
             ProductVersion productVersion = null;
-            productVersion = moduleVersion.metadata().map(Metadata::productVersion)
-                    .map(t -> {
-                        try {
-                            return t.getReference();
-                        } catch (RepositoryException e) {
-                            return null;
-                        }
-                    })
-                    .get();
+            if (moduleVersion.metadata().get().getValueMap().containsKey("productVersion")) {
+                productVersion = moduleVersion.metadata().map(Metadata::productVersion)
+                        .map(t -> {
+                            try {
+                                return t.getReference();
+                            } catch (RepositoryException e) {
+                                return null;
+                            }
+                        })
+                        .get();
+            }
+
             String productName = null;
             if (productVersion != null) {
-                productName = productVersion.getProduct().getValueMap().get("name").toString();
+                productName = productVersion.getProduct().name().get();
+            }
+            Calendar updatedDate = null;
+            if (moduleVersion.metadata().get().getValueMap().containsKey("dateUploaded")) {
```

Comment:
I think this can be simplified. The `map` calls are supposed to provide null-safety, but from your changes I suspect they aren’t at the end. The last `get()` call will throw a `NoSuchElementException` if the updated date is not present, so I suggest the following:

```suggestion
            Optional<Calendar> updatedDate = moduleVersion.metadata()
                    .map(Metadata::dataUploaded)
                    .map(Supplier::get);
```
And then in line 179 you should be able to do:
```java
            if(updatedDate.isPresent()) {
                arts.attribute(“pantheonupdateddate”,  dateFormat.format(updatedDate.get().getTime()));
            }
```

Reply:
updated!